### ISSUE TITLE
Allow import of custom inventory content

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#1420]
---

Author Name: **David Dimi** (David Dimi)
Original Redmine Issue: 2532, http://forge.fusioninventory.org/issues/2532
Original Date: 2014-06-02

---

Hi Fusioninventory Development Team,
a extended function of "additional-content" parameter or something similar is in fusioninventory-agent needed and would be very practical.

*For every server I need such information as:*
* Enabled Repositories 
(yum repolist -v | grep Repo-name | sed 's/^Repo-name\s*:\s*//g')
* Patchlevel of the server 
(yum repolist -v | grep Repo-name | sed 's/^Repo-name\s*:\s*//g')
* Full DMIDecode output or some types of the DMIDecode output, like dmidecode --type memory

To fullfill this requirements, Iam using the plugin called customfields (https://forge.indepnet.net/projects/customfields) to create required fields and then I have to maintain the fields manually, which costs much effort.
If fusioninventory-agent could take the work out my hand, it would be very nice and save me a lot of work.

*Like adding custom software information, as described in:*
http://www.fusioninventory.org/documentation/agent/additional_content/

I think not only I needs such feature, because everybody needs some custom informations.

In the attachment you can find some pictures how I setup the customfields plugin and how it looks like (customfields1.png and customfields2.png).

*Actually Iam using:*
* GLPI 0.84.5
* FusionInventory 0.84+2.2 and FusionInventory Agent (2.3.6)
* Custom Fields	1.6

If you need some more information or a tester, then don't hesitate to ask.

Many thanks in advance!

Best regards
David

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Guillaume Rousse** (@guillomovitch)
Original Date: 2014-06-02T14:18:01Z

---

Here are the steps needed to achieve this:

1) produce an arbitrary XML fragment through arbitrary command
For instance:
yum repolist -v | awk -F: 'BEGIN { print "<repositories>" } /Repo-name/ {print "<repository>"  $2 "</repository>"} END { print "</repositories>" }' > custom.xml

2) include this XML fragment in agent inventory output
This is possible through --additional-content switch, ie:
fusioninventory-agent --additional-content custom.xml

3) have the server recognize and process the information
This requires development on GLPI plugin side in order to inject this additional content into inventory, through custom import rules.

Basically, this is not an agent feature request, but a server one.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Dimi** (David Dimi)
Original Date: 2014-06-03T09:13:41Z

---

Feature request on GLPI side is made:
http://www.glpi-project.org/forum/viewtopic.php?id=36796

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Jean Dupont** (Jean Dupont)
Original Date: 2015-07-03T15:32:20Z

---

Allow me to second this feature (to be clear, I also added a request to the existing "GLPI forum thread":http://www.glpi-project.org/forum/viewtopic.php?id=36796 created by David Dimi).
